library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapter.java

```diff
@@ -793,6 +794,128 @@ void callSuperNotifyItemMoved(int fromPosition, int toPosition) {
+     * @param previousState previous state of section
+     */
+    public void notifyNotLoadedStateChanged(Section section, State previousState) {
+        State state = section.getState();
+
+        if (state == previousState) {
+            throw new IllegalStateException("No state changed");
+        }
+
+        if (previousState == State.LOADED) {
+            throw new IllegalStateException("Use notifyStateChangedFromLoaded");
+        }
+
+        if (state == State.LOADED) {
+            throw new IllegalStateException("Use notifyStateChangedToLoaded");
+        }
+
+        notifyItemChangedInSection(section, 0);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemChanged} and {@link #notifyItemInserted} with
+     * the position of the {@link State} view holder in the adapter. Useful to be called after
+     * changing the State from LOADING/ FAILED/ EMPTY to LOADED.
+     *
+     * @param tag unique identifier of the section
+     * @param previousState previous state of section
+     */
+    public void notifyStateChangedToLoaded(String tag, State previousState) {
+        Section section = getValidSectionOrThrowException(tag);
+
+        notifyStateChangedToLoaded(section, previousState);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemChanged} and {@link #notifyItemInserted} with
+     * the position of the {@link State} view holder in the adapter. Useful to be called after
+     * changing the State from LOADING/ FAILED/ EMPTY to LOADED.
+     *
+     * @param section a visible section of this adapter
+     * @param previousState previous state of section
+     */
+    public void notifyStateChangedToLoaded(Section section, State previousState) {
+        State state = section.getState();
+
+        if (state == previousState) {
+            throw new IllegalStateException("No state changed");
+        }
+
+        if (state != State.LOADED) {
+            if (previousState == State.LOADED) {
+                throw new IllegalStateException("Use notifyStateChangedFromLoaded");
+            } else {
+                throw new IllegalStateException("Use notifyNotLoadedStateChanged");
+            }
+        }
+
+        notifyItemChangedInSection(section, 0);
+
+        int contentItemsTotal = section.getContentItemsTotal();
+
+        if (contentItemsTotal > 1) {
+            notifyItemRangeInsertedInSection(section, 1, contentItemsTotal - 1);
+        }
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemRangeRemoved} and {@link #notifyItemChanged} with
+     * the position of the {@link State} view holder in the adapter. Useful to be called after
+     * changing the State from LOADED to LOADING/ FAILED/ EMPTY.
+     *
+     * @param tag unique identifier of the section
+     * @param previousContentItemsCount previous content items count of section
+     */
+    public void notifyStateChangedFromLoaded(String tag, int previousContentItemsCount) {
```

Comment:
@luizgrp 

Just wondering, instead of using `int previousContentItemsCount`, should it be named as `int previousContentItemsTotal` ?

As, we have Section method named `getContentItemsTotal()`.